Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm

```diff
@@ -1418,6 +1418,14 @@ - (void)_handleWebPushAction:(_WKWebPushAction *)webPushAction
     }
 }
 
+- (void)_runningOrTerminatingServiceWorkerCountForTesting:(void(^)(NSUInteger))completionHandler
+{
+    auto completionHandlerCopy = makeBlockPtr(completionHandler);
+    _websiteDataStore->runningOrTerminatingServiceWorkerCountForTesting([completionHandlerCopy = WTFMove(completionHandlerCopy)](auto result) {
+        completionHandlerCopy(result);
```

Comment:
Maybe it would be useful for UIProcess to know whether a service worker process has any running service worker. That way, this process could exit in case of memory pressure. Or maybe the WebProcess should do this on its own, if it is not already happening.

Reply:
Yeah I was surprised there wasn't already a good way of finding this state without messaging NetworkProcess. But will look in to it in the future to not complicate this patch any further.